server/app/services/scheduler/strategy/random.rb

```diff
@@ -13,22 +13,32 @@ def host_grace_period
         30.seconds
       end
 
-      ##
       # @param [GridService] grid_service
       # @param [Integer] instance_number
-      # @param [Array<HostNode>] nodes
+      # @param [Array<Scheduler::Node>] nodes
+      # @return [Scheduler::Node, NilClass]
       def find_node(grid_service, instance_number, nodes)
         if grid_service.stateless?
+          node = find_previous_or_new_node(grid_service, instance_number, nodes)
+          return node if node
```

Comment:
It's not exactly the same.. for stateful we cannot sample if previous instance exists but node cannot be found from nodes list. Here we can always sample if `find_previous_or_new_node` returns `nil`.